embedded-graphics/Cargo.toml

```diff
@@ -34,3 +34,4 @@ nalgebra_support = [ "nalgebra" ]
 [dev-dependencies]
 arrayvec = "0.5.1"
 tinytga = { version = "0.3.1", features = [ "graphics" ] }
+embedded-graphics-simulator = "0.2.0-beta.2"
```

Comment:
I got this to "work" with:

```toml
[target.'cfg(target_arch = "x86_64")'.dependencies]
embedded-graphics-simulator = "0.2.0-beta.2"
```

However now Rust complains about a cyclic dependency :(

Reply:
```
[target.'cfg(target_arch = "x86_64")'.dev-dependencies]
embedded-graphics-simulator = "0.2.0-beta.2"
```
gets rid of the cyclic dependency, but somehow tries to build some crates with `std` support, even if the arch doesn't match.

Any ideas how to solve this? Checking the e-g README while building the simulator could work, but is a hack.

Reply:
Seems to be the `--all-features` flag in `build_target.sh`. Try changing `set -e` to set -ex` at the top and it'll show the bailing command. I can repro if I run this:

```bash
cargo build -p embedded-graphics --target thumbv7m-none-eabi --features nalgebra_support
```

This builds fine:

```bash
cargo build -p embedded-graphics --target thumbv7m-none-eabi
```

Very weird because this works in master 🤔 

> Any ideas how to solve this? Checking the e-g README while building the simulator could work, but is a hack.

We could add another, unpublished crate to the workspace that pulls in all the readme(s) as examples. Would give a bit more control I think, but yeah still feels quite hacky.

Reply:
Enabling the new feature resolver (https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#features) seems to fix this problem. But until this gets stabilized we need another solution, so I decided to move all README checks into the simulator crate for now. Are you OK with leaving it like this?

Reply:
Yeah that's fine, as long as they get tested in CI at some point that's cool. I don't think a `cargo publish` runs tests so we're not missing any safety there. Is it worth opening a tracking issue for reverting the README move when the new resolver stabilised? If you think so, please open one, otherwise it works as is so let's leave it.